binutils.mk

```diff
@@ -0,0 +1,161 @@
+
+BINUTILS_CONFARGS := --enable-obsolete \
+	--enable-shared \
+	--enable-plugins \
+	--enable-threads \
+	--with-system-zlib \
+	--enable-deterministic-archives \
+	--disable-compressed-debug-sections \
+	--enable-new-dtags \
+	--disable-x86-used-note \
+	--with-pkgversion="GNU Binutils for Procursus" \
+	--enable-ld=default \
+	--enable-gold \
+	--enable-default-hash-style=gnu
+
+binutils-setup: setup
+	wget -q -nc -P $(BUILD_SOURCE) https://ftpmirror.gnu.org/binutils/binutils-$(BINUTILS_VERSION).tar.xz{,.sig}
+	$(call PGP_VERIFY,binutils-$(BINUTILS_VERSION).tar.xz)
+	mkdir -p $(BUILD_WORK)/binutils
+	$(foreach target,$(BINUTILS_TARGETS),$(call EXTRACT_TAR,binutils-$(BINUTILS_VERSION).tar.xz,binutils-$(BINUTILS_VERSION),binutils/$(target))
+	$(foreach target,$(BINUTILS_TARGETS),$(call DO_PATCH,binutils,binutils/$(target),-p1))
+
+ifneq ($(wildcard $(BUILD_WORK)/binutils/.build_complete),)
+binutils:
+	@echo "Using previously built binutils."
+else
+binutils: binutils-setup gettext
+	for target in $(BINUTILS_TARGETS); do \
+		cd $(BUILD_WORK)/binutils/$$target && ./configure -C \
+			--build=$$($(BUILD_MISC)/config.guess) \
+			--host=$(GNU_HOST_TRIPLE) \
+			--prefix=$(MEMO_PREFIX)$(MEMO_SUB_PREFIX) \
+			--libdir=$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/$(GNU_HOST_TRIPLE)/$$target/lib \
+			--includedir=$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/$(GNU_HOST_TRIPLE)/$$target/include \
+			--target=$$target \
+			$(BINUTILS_CONFARGS); \
+		$(MAKE) -C $(BUILD_WORK)/binutils/$$target; \
+		$(MAKE) -C $(BUILD_WORK)/binutils/$$target install \
+			DESTDIR=$(BUILD_STAGE)/binutils/$$target; \
+	done
+	touch $(BUILD_WORK)/binutils/.build_complete
+endif
+
+binutils-package: binutils-stage
+	# binutils.mk Package Structure
+	for target in $(BINUTILS_TARGETS); do \
+		rm -rf $(BUILD_DIST)/binutils-$$target; \
+	done
+	rm -rf $(BUILD_DIST)/binutils-common
+	mkdir -p $(BUILD_DIST)/binutils-common/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)
+	
+	# binutils.mk Prep binutils-*
+	for target in $(BINUTILS_TARGETS); do \
+		cp -r $(BUILD_STAGE)/binutils/$$target $(BUILD_DIST)/binutils-$$target; \
+	done
+	for target in $(BINUTILS_TARGETS); do \
+		for pkg in $(BUILD_DIST)/binutils-$$target; do \
+			for man in $$pkg/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/share/man/man1/*; do \
+				$(LN) -sf binutils-$$(basename $$man .1 | rev | cut -d- -f1 | rev ).1.zst $$man; \
+			done; \
+			rm -rf $$pkg/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/share/{locale,info}; \
+		done; \
+	done
+	cp -r $(BUILD_STAGE)/binutils/x86_64-linux-gnu/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/share $(BUILD_DIST)/binutils-common/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)
+	for man in $(BUILD_DIST)/binutils-common/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/share/man/man1/*; do \
+		mv $$man $(BUILD_DIST)/binutils-common/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/share/man/man1/binutils-$$(basename $$man .1 | rev | cut -d- -f1 | rev ).1; \
+	done
+	
+	# binutils.mk Sign
+	$(call SIGN,binutils-aarch64-linux-gnu,general.xml)
```

Comment:
Would doing a for $$target loop in the SIGN and PACK stages decrease the size of this makefile? Also, great to see you're still alive. :)